aggregate.go

```diff
@@ -0,0 +1,132 @@
+package versionbundle
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/giantswarm/microerror"
+)
+
+// Aggregate merges capabilities based on dependencies version bundles within
+// the given capabilities define for their components.
+func Aggregate(capabilities []Capability) (Aggregation, error) {
+	err := ValidateCapabilities(capabilities).Validate()
+	if err != nil {
+		return Aggregation{}, microerror.Mask(err)
+	}
+
+	var newAggregration Aggregation
+
+	if len(capabilities) == 0 {
+		return newAggregration, nil
+	}
+
+	if len(capabilities) == 1 {
+		newAggregration.Capabilities = append(newAggregration.Capabilities, []Capability{capabilities[0]})
```

Comment:
Why not just 

```
newAggregation.Capabilities = capabilties
```
?

Reply:
Because the aggregation is a list of lists while the input list is just a list.

Reply:
But `[]Capability{capabilities[0]}` is the same as `capabilities`.

Reply:
Done.